pkg/controllermanager/controller/project/project_activity_control_test.go

```diff
@@ -0,0 +1,186 @@
+			},
+			Spec: gardencorev1beta1.ProjectSpec{
+				Namespace: &namespaceName,
+			},
+		}
+		shoot = &gardencorev1beta1.Shoot{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:              shootName,
+				Namespace:         namespaceName,
+				UID:               "1",
+				CreationTimestamp: metav1.Time{Time: time.Date(1, 2, 1, 1, 1, 1, 1, time.UTC)},
+			},
+		}
+		shootWithoutProject = &gardencorev1beta1.Shoot{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:              shootName,
+				Namespace:         "fake",
+				UID:               "1",
+				CreationTimestamp: metav1.Time{Time: time.Date(1, 3, 1, 1, 1, 1, 1, time.UTC)},
+			},
+		}
+		errorShoot = &gardencorev1beta1.Shoot{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:              shootName,
+				Namespace:         "error",
+				UID:               "1",
+				CreationTimestamp: metav1.Time{Time: time.Date(1, 4, 1, 1, 1, 1, 1, time.UTC)},
+			},
+		}
+
+		logger.Logger = logger.NewLogger("info")
+		ctrl = gomock.NewController(GinkgoT())
+		k8sGardenRuntimeClient = mockclient.NewMockClient(ctrl)
+		reconciler = NewActivityReconciler(logger.NewNopLogger(), k8sGardenRuntimeClient)
+		request = reconcile.Request{NamespacedName: types.NamespacedName{Name: shoot.Name, Namespace: shoot.Namespace}}
+		errorRequest = reconcile.Request{NamespacedName: types.NamespacedName{Name: errorShoot.Name, Namespace: errorShoot.Namespace}}
+		k8sGardenRuntimeClient.EXPECT().List(ctx, gomock.AssignableToTypeOf(&gardencorev1beta1.ProjectList{}), gomock.Any()).DoAndReturn(func(_ context.Context, obj *gardencorev1beta1.ProjectList, opts client.MatchingFields) error {
+			if reflect.DeepEqual(opts[core.ProjectNamespace], *project.Spec.Namespace) {
+				*obj = gardencorev1beta1.ProjectList{Items: []gardencorev1beta1.Project{*project}}
+				return nil
+			}
+			if reflect.DeepEqual(opts[core.ProjectNamespace], "error") {
+				return errors.New("API ERROR")
+			}
+			logger.Logger.Infof("Project %s not found returning empty", opts[core.ProjectNamespace])
+			*obj = gardencorev1beta1.ProjectList{}
+			return nil
+		})
+
+		k8sGardenRuntimeClient.EXPECT().Get(ctx, gomock.Any(), gomock.AssignableToTypeOf(&gardencorev1beta1.Shoot{})).DoAndReturn(func(_ context.Context, namespacedName client.ObjectKey, obj *gardencorev1beta1.Shoot) error {
+			for _, s := range []gardencorev1beta1.Shoot{*shoot, *shootWithoutProject, *errorShoot} {
+				if reflect.DeepEqual(namespacedName.Name, s.Name) && reflect.DeepEqual(namespacedName.Namespace, s.Namespace) {
+					*obj = s
+					return nil
+				}
+			}
+			return apierrors.NewNotFound(gardencorev1beta1.Resource("Project"), "<unknown>")
+		})
+
+		k8sGardenRuntimeClient.EXPECT().List(ctx, gomock.AssignableToTypeOf(&gardencorev1beta1.ShootList{}), gomock.Any()).DoAndReturn(func(_ context.Context, obj *gardencorev1beta1.ShootList, _ ...client.ListOption) error {
+			obj.Items = []gardencorev1beta1.Shoot{*shoot, *shootWithoutProject, *errorShoot}
+
+			return nil
+		})
```

Comment:
This is expected only if the last activity timestamp is not set, otherwise there will be no list shoots requests.